search-service/src/main/kotlin/com/egm/stellio/search/authorization/permission/service/PermissionService.kt

```diff
@@ -0,0 +1,416 @@
+    private val databaseClient: DatabaseClient,
+    private val r2dbcEntityTemplate: R2dbcEntityTemplate,
+    private val applicationProperties: ApplicationProperties,
+    private val subjectReferentialService: SubjectReferentialService,
+) {
+
+    @Transactional
+    suspend fun create(
+        permission: Permission,
+    ): Either<APIException, Unit> = either {
+        permission.validate().bind()
+        checkExistence(permission.id, true).bind()
+
+        val insertStatement =
+            """
+            INSERT INTO permission(
+                id,
+                target_id,
+                assignee,
+                action,
+                created_at,
+                modified_at,
+                assigner
+            )
+            VALUES(
+                :id,
+                :target_id,
+                :assignee,
+                :action,
+                :created_at,
+                :modified_at,
+                :assigner
+            )
+            """.trimIndent()
+        databaseClient.sql(insertStatement)
+            .bind("id", permission.id)
+            .bind("target_id", permission.target.id)
+            .bind("assignee", permission.assignee)
+            .bind("action", permission.action.value)
+            .bind("created_at", permission.createdAt)
+            .bind("modified_at", permission.modifiedAt)
+            .bind("assigner", permission.assigner)
+            .execute().bind()
+    }
+
+    suspend fun checkExistence(
+        id: URI,
+        inverse: Boolean = false
+    ): Either<APIException, Unit> =
+        databaseClient.sql(
+            """
+            SELECT exists (
+                SELECT 1
+                FROM permission
+                WHERE id = :id
+            ) as exists
+            """.trimIndent()
+        ).bind("id", id)
+            .oneToResult { toBoolean(it["exists"]) }
+            .flatMap {
+                if (it && !inverse)
+                    Unit.right()
+                else if (!it && inverse)
+                    Unit.right()
+                else if (it)
+                    AlreadyExistsException(Permission.alreadyExistsMessage(id)).left()
+                else
+                    ResourceNotFoundException(Permission.notFoundMessage(id)).left()
+            }
+
+    suspend fun getById(id: URI): Either<APIException, Permission> = either {
+        checkExistence(id).bind()
+
+        val selectStatement =
+            """
+            SELECT id,
+                target_id,
+                assignee,
+                action,
+                created_at,
+                modified_at,
+                assigner
+            FROM permission  
+            WHERE id = :id
+            """.trimIndent()
+
+        return databaseClient.sql(selectStatement)
+            .bind("id", id)
+            .oneToResult { rowToPermission(it).bind() }
+    }
+
+    suspend fun isAdminOf(id: URI, sub: Option<Sub>): Either<APIException, Boolean> = either {
```

Comment:
sometimes, `id` is the id of the permission, other times it is the `id` of the target entity id. this is confusing and makes the code more difficult to understand.

I suggest to use `id` for the permission (in the context of the `PermissionService`) and `entityId` when it refers to target entity id.